src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java

```diff
@@ -201,6 +202,31 @@ public void testCrossCheckRGs(final File file1, final File file2, final boolean
         doTest(args.toArray(new String[0]), metrics, expectedRetVal, expectedNMetrics, CrosscheckMetric.DataType.READGROUP, expectAllMatch);
     }
 
+    @Test(dataProvider = "bamFilesRGs")
+    public void testCrossCheckRGs_with_short_dictionary(final File file1, final File file2, final boolean expectAllMatch, final int expectedRetVal, final int expectedNMetrics) throws IOException {
+
+        File metrics = File.createTempFile("Fingerprinting", "NA1291.RG.crosscheck_metrics");
+        metrics.deleteOnExit();
+
+        final List<String> args = new ArrayList<>(Arrays.asList("INPUT=" + file1.getAbsolutePath(),
+                "INPUT=" + file2.getAbsolutePath(),
+                "OUTPUT=" + metrics.getAbsolutePath(),
+                "LOD_THRESHOLD=" + -2.0,
+                "EXPECT_ALL_GROUPS_TO_MATCH=" + expectAllMatch)
+        );
+
+        if (file1.getName().endsWith(SamReader.Type.CRAM_TYPE.fileExtension())) {
+            args.add("R=" + referenceForCrams);
+            args.add("HAPLOTYPE_MAP=" + HAPLOTYPE_MAP_FOR_CRAMS);
```

Comment:
could you add a HAPLOTYPE_MAP_SHORT_FOR_CRAMS as well just so we're testing that the functionality works for both bams and crams?

Reply:
done.